pinot-core/src/main/java/org/apache/pinot/core/startree/StarTreeUtils.java

```diff
@@ -48,14 +48,11 @@ public class StarTreeUtils {
   private StarTreeUtils() {
   }
 
-  public static final String USE_STAR_TREE_KEY = "useStarTree";
-
   /**
    * Returns whether star-tree is disabled for the query.
    */
   public static boolean isStarTreeDisabled(QueryContext queryContext) {
-    Map<String, String> debugOptions = queryContext.getDebugOptions();
-    return debugOptions != null && "false".equalsIgnoreCase(debugOptions.get(USE_STAR_TREE_KEY));
+    return queryContext.isSkipStartree();
```

Comment:
This is backward-incompatible change. Suggest still supporting the old way of skipping star-tree so that user can put both options during the migration, then we may deprecate it in the future.
We can apply this logic in `InstancePlanMakerImplV2.applyQueryOptions()`. If either of the flag is set, we should skip the star-tree